src/go/rdctl/cmd/start.go

```diff
@@ -0,0 +1,178 @@
+	}
+	if applicationPath == "" {
+		pathLookupFuncs := map[string]func() string{
+			"windows": getWindowsRDPath,
+			"linux":   getLinuxRDPath,
+			"darwin":  getMacOSRDPath,
+		}
+		getPathFunc, ok := pathLookupFuncs[runtime.GOOS]
+		if !ok {
+			return fmt.Errorf("don't know how to find the path to Rancher Desktop on OS %s", runtime.GOOS)
+		}
+		applicationPath = getPathFunc()
+		if applicationPath == "" {
+			return fmt.Errorf("no executable found in the default location; please retry with the --path|-p option")
+		}
+	}
+	return launchApp(applicationPath, commandLineArgs)
+}
+
+func launchApp(applicationPath string, commandLineArgs []string) error {
+	var commandName string
+	var args []string
+
+	if runtime.GOOS == "darwin" {
+		commandName = "open"
+		args = append(args, "-a", applicationPath)
+		if len(commandLineArgs) > 0 {
+			args = append(args, "--args")
+			args = append(args, commandLineArgs...)
+		}
+	} else {
+		commandName = applicationPath
+		args = commandLineArgs
+	}
+	// Include this output because there's a delay before the UI comes up.
+	// Without this line, it might look like the command doesn't work.
+	fmt.Fprintf(os.Stderr, "About to launch %s %s ...\n", commandName, strings.Join(args, " "))
+	cmd := exec.Command(commandName, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func getWindowsRDPath() string {
+	localAppDataDir := os.Getenv("LOCALAPPDATA")
+	if localAppDataDir == "" {
+		var homeDir string
+		homeDrive := os.Getenv("HOMEDRIVE")
+		homePath := os.Getenv("HOMEPATH")
+		if homeDrive != "" && homePath != "" {
+			homeDir = homeDrive + homePath
+		} else {
+			homeDir = os.Getenv("HOME")
+		}
+		if homeDir == "" {
+			return ""
+		}
+		localAppDataDir = path.Join(homeDir, "AppData", "Local")
+	}
+	return checkExistence(path.Join(localAppDataDir, "Programs", "Rancher Desktop", "Rancher Desktop.exe"), 0)
+}
+
+func getMacOSRDPath() string {
+	return checkExistence(path.Join("/Applications", "Rancher Desktop.app"), 0)
+}
+
+func getLinuxRDPath() string {
+	candidatePath := checkExistence("/opt/rancher-desktop/rancher-desktop", 0111)
+	if candidatePath != "" {
+		return candidatePath
+	}
+	var err error
+	candidatePath, err = exec.LookPath("rancher-desktop")
+	if err != nil {
+		return ""
+	}
+	// No need to check existence or mode bits for this file because `exec.LookPath` did that for us.
+	return candidatePath
+}
+
+/**
+ * Verify the path exists. For Linux pass in mode bits to guarantee the file is executable (for at least one
+ * category of user). Note that on macOS the candidate is a directory, so never pass in mode bits.
+ * And mode bits don't make sense on Windows.
+ */
+func checkExistence(candidatePath string, modeBits uint32) string {
+	stat, err := os.Stat(candidatePath)
+	if err != nil {
+		return ""
+	}
+	if modeBits != 0 && (!stat.Mode().IsRegular() || (uint32(stat.Mode().Perm())&modeBits == 0)) {
```

Comment:
I think letting `modeBits` be `fs.FileMode` rather than casting here is clearer. (A literal `0111` is untyped, so we don't need a cast anywhere.)

Reply:
True. I'm also removing the extra parens around `(A&B == C)` because unlike C/C++, go got the precedence for the bit ops correct (they're the same as mult/divide/modulus).